packages/esm-patient-medications-app/src/add-drug-order/drug-order-form.component.tsx

```diff
@@ -675,31 +705,55 @@ const CustomNumberInput = ({ setValue, control, name, labelText, ...inputProps }
   );
 };
 
-const ControlledFieldInput = ({ name, control, type, ...restProps }) => {
+interface ControlledFieldInputProps {
+  name: keyof MedicationOrderFormData;
+  type: 'toggle' | 'checkbox' | 'number' | 'textArea' | 'textInput' | 'comboBox';
+  handleAfterChange?: (
+    newValue: MedicationOrderFormData[keyof MedicationOrderFormData],
+    prevValue: MedicationOrderFormData[keyof MedicationOrderFormData],
+  ) => void;
+  control: Control<MedicationOrderFormData>;
+  [x: string]: any;
+}
+
+const ControlledFieldInput = ({
+  name,
+  type,
+  control,
+  watch,
+  handleAfterChange,
+  ...restProps
+}: ControlledFieldInputProps) => {
   const {
     field: { onBlur, onChange, value, ref },
     fieldState,
   } = useController<MedicationOrderFormData>({ name: name, control });
 
+  const handleChange = (newValue: MedicationOrderFormData[keyof MedicationOrderFormData]) => {
+    const prevValue = watch?.(name);
```

Comment:
Again, for this purpose, `getValues()` is a better choice than `watch`. This should also be wrapped in a `useCallback()` hook.